download-maven-plugin/src/main/java/com/googlecode/download/maven/plugin/internal/cache/Index.java

```diff
@@ -0,0 +1,32 @@
+package com.googlecode.download.maven.plugin.internal.cache;
+
+/**
+ * Cache index.
+ * @author Paul Polishchuk
+ * @since 1.3.1
+ */
+interface Index {
+
+    /**
+     * Adds given path to the index using url parameter as a key.
+     * @param url index key
+     * @param path index value
+     */
+    void put(String url, String path);
```

Comment:
The index should most likely check for existence of the File, using new File(path).isFile(). So it makes sense to directly pass the File.
However, if you don't want to do it right now, I can merge you patch without it, and I will change it later ;)